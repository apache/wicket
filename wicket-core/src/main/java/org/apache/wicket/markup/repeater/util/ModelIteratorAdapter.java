/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.wicket.markup.repeater.util;

import java.util.Iterator;

import jakarta.annotation.Nonnull;

import org.apache.wicket.markup.repeater.RefreshingView;
import org.apache.wicket.model.IModel;
import org.apache.wicket.util.lang.Args;


/**
 * Iterator adapter that wraps adaptee's elements with a model. Convenient when implementing
 * {@link RefreshingView}.
 * 
 * @author Igor Vaynberg (ivaynberg)
 * @param <T>
 * 
 */
public abstract class ModelIteratorAdapter<T> implements Iterator<IModel<T>>
{
	private final Iterator<T> delegate;

	/**
	 * Constructor
	 * 
	 * @param iterable
	 *            iterable whose iterator will be wrapped
	 */
	public ModelIteratorAdapter(@Nonnull Iterable<T> iterable)
	{
		Args.notNull(iterable, "iterable");
		this.delegate = iterable.iterator();
	}


	/**
	 * Constructor
	 * 
	 * @param delegate
	 *            iterator that will be wrapped
	 */
	public ModelIteratorAdapter(Iterator<T> delegate)
	{
		this.delegate = delegate;
	}

	/**
	 * @see java.util.Iterator#hasNext()
	 */
	@Override
	public boolean hasNext()
	{
		return delegate.hasNext();
	}

	/**
	 * @see java.util.Iterator#next()
	 */
	@Override
	public IModel<T> next()
	{
		return model(delegate.next());
	}

	/**
	 * @see java.util.Iterator#remove()
	 */
	@Override
	public void remove()
	{
		delegate.remove();
	}

	/**
	 * This method is used to wrap the provided object with an implementation of IModel. The
	 * provided object is guaranteed to be returned from the delegate iterator.
	 * 
	 * @param object
	 *            object to be wrapped
	 * @return IModel wrapper for the object
	 */
	abstract protected IModel<T> model(T object);
}
