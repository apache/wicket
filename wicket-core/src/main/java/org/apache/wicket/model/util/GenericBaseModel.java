/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.apache.wicket.model.util;

import java.io.Serializable;

import org.apache.wicket.model.IDetachable;
import org.apache.wicket.model.IObjectClassAwareModel;
import org.apache.wicket.util.lang.Objects;

/**
 * Base class for models that contain instances that do not appear to be serializable and cannot
 * thus use <code>Model</code> directly. Makes sure that the object set into the model can be
 * serialized.
 * 
 * @author Timo Rantalaiho
 * @param <T>
 *            type of model object
 */
public abstract class GenericBaseModel<T> implements IObjectClassAwareModel<T>
{
	private static final long serialVersionUID = 1L;
	/** model object */
	private T object;

	@Override
	public T getObject()
	{
		return object;
	}

	/**
	 * Set the model object. The contents must be serializable, as they are stored in the session
	 * 
	 * @param object
	 *            the model object
	 * @see org.apache.wicket.model.IModel#setObject(Object)
	 */
	@Override
	public void setObject(T object)
	{
		if (!(object instanceof Serializable))
		{
			object = createSerializableVersionOf(object);
		}
		this.object = object;
	}

	/**
	 * Creates a serializable version of the object. The object is usually a collection.
	 * 
	 * @param object
	 * @return serializable version of <code>object</code>
	 */
	protected abstract T createSerializableVersionOf(T object);

	@Override
	public void detach()
	{
		if (object instanceof IDetachable)
		{
			((IDetachable)object).detach();
		}
	}

	@SuppressWarnings("unchecked")
	@Override
	public Class<T> getObjectClass()
	{
		return object != null ? (Class<T>) object.getClass() : null;
	}

	@Override
	public String toString()
	{
	 StringBuilder sb = new StringBuilder("Model:classname=[");
		sb.append(getClass().getName()).append(']');
		sb.append(":object=[").append(object).append(']');
		return sb.toString();
	}

	@Override
	public int hashCode()
	{
		return Objects.hashCode(object);
	}

	@Override
	public boolean equals(Object obj)
	{
		if (this == obj)
		{
			return true;

		}
		if (!(obj instanceof GenericBaseModel))
		{
			return false;
		}
		GenericBaseModel<?> that = (GenericBaseModel<?>)obj;
		return Objects.equal(object, that.object);
	}
}
