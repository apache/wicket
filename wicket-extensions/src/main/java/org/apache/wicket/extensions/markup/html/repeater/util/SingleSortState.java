/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.wicket.extensions.markup.html.repeater.util;

import org.apache.wicket.extensions.markup.html.repeater.data.sort.ISortState;
import org.apache.wicket.extensions.markup.html.repeater.data.sort.SortOrder;
import org.apache.wicket.util.io.IClusterable;
import org.apache.wicket.util.lang.Args;

/**
 * Implementation of ISortState that can keep track of sort information for a single property.
 * 
 * @author Igor Vaynberg (ivaynberg)
 * @param <T>
 *            the type of the sort property
 * 
 */
public class SingleSortState<T> implements ISortState<T>, IClusterable
{
	private static final long serialVersionUID = 1L;

	SortParam<T> param;

	@Override
	public void setPropertySortOrder(final T property, final SortOrder order)
	{
		Args.notNull(property, "property");
		Args.notNull(order, "order");

		if (order == SortOrder.NONE)
		{
			if ((param != null) && property.equals(param.getProperty()))
			{
				param = null;
			}
		}
		else
		{
			param = new SortParam<T>(property, order == SortOrder.ASCENDING);
		}
	}

	@Override
	public SortOrder getPropertySortOrder(final T property)
	{
		Args.notNull(property, "property");

		if ((param == null) || (param.getProperty().equals(property) == false))
		{
			return SortOrder.NONE;
		}
		return param.isAscending() ? SortOrder.ASCENDING : SortOrder.DESCENDING;
	}

	/**
	 * @return current sort state
	 */
	public SortParam<T> getSort()
	{
		return param;
	}

	/**
	 * Sets the current sort state
	 * 
	 * @param param
	 *            parameter containing new sorting information
	 */
	public void setSort(final SortParam<T> param)
	{
		this.param = param;
	}

	@Override
	public String toString()
	{
		return "[SingleSortState sort=" + ((param == null) ? "null" : param.toString()) + ']';
	}

}
