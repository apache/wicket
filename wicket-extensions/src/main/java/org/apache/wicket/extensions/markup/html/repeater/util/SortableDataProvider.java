/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.wicket.extensions.markup.html.repeater.util;

import org.apache.wicket.extensions.markup.html.repeater.data.sort.ISortState;
import org.apache.wicket.extensions.markup.html.repeater.data.sort.SortOrder;
import org.apache.wicket.extensions.markup.html.repeater.data.table.ISortableDataProvider;


/**
 * Convenience implementation of a data provider that can also act as a locator for a
 * {@link SingleSortState} object.
 * 
 * Most times it is convenient to keep sort and filtering information inside the data provider
 * implementation because it makes that information easy to access within the data provider.
 * 
 * @author Igor Vaynberg (ivaynberg at apache dot org)
 * @param <T>
 * @param <S>
 *     the type of the sorting parameter
 */
public abstract class SortableDataProvider<T, S> implements ISortableDataProvider<T, S>
{
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	private SingleSortState<S> state = new SingleSortState<S>();

	/**
	 * @see ISortableDataProvider#getSortState()
	 */
	@Override
	public final ISortState<S> getSortState()
	{
		return state;
	}

	/**
	 * Returns current sort state
	 * 
	 * @return current sort state
	 */
	public SortParam<S> getSort()
	{
		return state.getSort();
	}

	/**
	 * Sets the current sort state
	 * 
	 * @param param
	 *            parameter containing new sorting information
	 */
	public void setSort(final SortParam<S> param)
	{
		state.setSort(param);
	}

	/**
	 * Sets the current sort state
	 * 
	 * @param property
	 *            sort property
	 * @param order
	 *            sort order
	 */
	public void setSort(final S property, final SortOrder order)
	{
		state.setPropertySortOrder(property, order);
	}

	/**
	 * @see ISortableDataProvider#detach()
	 */
	@Override
	public void detach()
	{
	}


}
