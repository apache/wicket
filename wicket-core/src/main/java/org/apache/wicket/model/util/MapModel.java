/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.wicket.model.util;

import java.util.HashMap;
import java.util.Map;


/**
 * Based on <code>Model</code> but for maps of serializable objects.
 * 
 * @author Timo Rantalaiho
 * @param <K>
 *            map's key type
 * @param <V>
 *            map's value type
 */
public class MapModel<K, V> extends GenericBaseModel<Map<K, V>>
{
	private static final long serialVersionUID = 1L;

	/**
	 * Creates empty model
	 */
	public MapModel()
	{
	}

	/**
	 * Creates model that will contain <code>map</code>
	 * 
	 * @param map
	 */
	public MapModel(Map<K, V> map)
	{
		setObject(map);
	}

	@Override
	protected Map<K, V> createSerializableVersionOf(Map<K, V> object)
	{
		return new HashMap<>(object);
	}
}
