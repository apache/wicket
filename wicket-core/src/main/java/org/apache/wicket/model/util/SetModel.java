/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.wicket.model.util;

import java.util.HashSet;
import java.util.Set;


/**
 * Based on <code>Model</code> but for sets of serializable objects.
 * 
 * @author Timo Rantalaiho
 * @param <T>
 *            type of object inside set
 */
public class SetModel<T> extends GenericBaseModel<Set<T>>
{
	private static final long serialVersionUID = 1L;

	/**
	 * Creates empty model
	 */
	public SetModel()
	{
	}

	/**
	 * Creates model that will contain <code>set</code>
	 * 
	 * @param set
	 */
	public SetModel(Set<T> set)
	{
		setObject(set);
	}

	@Override
	protected Set<T> createSerializableVersionOf(Set<T> object)
	{
		return new HashSet<>(object);
	}
}
