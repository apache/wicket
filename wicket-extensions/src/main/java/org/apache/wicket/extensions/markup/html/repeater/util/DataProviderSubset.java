/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.wicket.extensions.markup.html.repeater.util;

import org.apache.wicket.markup.repeater.data.IDataProvider;
import org.apache.wicket.model.IDetachable;
import org.apache.wicket.model.IModel;

import java.util.Collection;
import java.util.HashSet;
import java.util.Iterator;
import java.util.Set;

/**
 * A {@link Set} implementation utilizing a {@link IDataProvider}'s models to keep containing
 * elements.
 *
 * @param <T> type of data
 * @author Roland Kurucz
 * @see IDataProvider#model(Object)
 */
public class DataProviderSubset<T> implements Set<T>, IDetachable {

    private static final long serialVersionUID = 1L;

    private final IDataProvider<T> provider;

    private final Set<IModel<T>> models = new HashSet<>(0);

    /**
     * Create an empty subset.
     *
     * @param provider the provider of the complete set
     */
    public DataProviderSubset(IDataProvider<T> provider) {
        this.provider = provider;
    }

    @Override
    public void detach() {
        for (IModel<T> model : models) {
            model.detach();
        }
    }

    @Override
    public int size() {
        return models.size();
    }

    @Override
    public boolean isEmpty() {
        return models.size() == 0;
    }

    @Override
    public void clear() {
        detach();

        models.clear();
    }

    @Override
    public boolean contains(Object o) {
        IModel<T> model = model(o);

        boolean contains = models.contains(model);

        model.detach();

        return contains;
    }

    @Override
    public boolean add(T t) {
        return models.add(model(t));
    }

    @Override
    public boolean remove(Object o) {
        IModel<T> model = model(o);

        boolean removed = models.remove(model);

        model.detach();

        return removed;
    }

    @Override
    public Iterator<T> iterator() {
        return new Iterator<T>() {

            private Iterator<IModel<T>> iterator = models.iterator();

            private IModel<T> current;

            @Override
            public boolean hasNext() {
                return iterator.hasNext();
            }

            @Override
            public T next() {
                current = iterator.next();

                return current.getObject();
            }

            @Override
            public void remove() {
                iterator.remove();

                current.detach();
                current = null;
            }
        };
    }

    @Override
    public boolean addAll(Collection<? extends T> ts) {
        boolean changed = false;

        for (T t : ts) {
            changed |= add(t);
        }

        return changed;
    }

    @Override
    public boolean containsAll(Collection<?> cs) {
        for (Object c : cs) {
            if (!contains(c)) {
                return false;
            }
        }
        return true;
    }

    @Override
    public boolean removeAll(Collection<?> cs) {
        boolean changed = false;

        for (Object c : cs) {
            changed |= remove(c);
        }

        return changed;
    }

    @Override
    public boolean retainAll(Collection<?> c) {
        throw new UnsupportedOperationException();
    }

    @Override
    public Object[] toArray() {
        throw new UnsupportedOperationException();
    }

    @Override
    public <S> S[] toArray(S[] a) {
        throw new UnsupportedOperationException();
    }

    @SuppressWarnings("unchecked")
    private IModel<T> model(Object o) {
        return provider.model((T) o);
    }

    /**
     * Create a model holding this set.
     *
     * @return model
     */
    public IModel<Set<T>> createModel() {
        return new IModel<>() {
            private static final long serialVersionUID = 1L;

            @Override
            public Set<T> getObject() {
                return DataProviderSubset.this;
            }

            @Override
            public void detach() {
                DataProviderSubset.this.detach();
            }
        };
    }
}
