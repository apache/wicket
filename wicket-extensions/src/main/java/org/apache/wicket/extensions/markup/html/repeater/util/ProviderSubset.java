/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.wicket.extensions.markup.html.repeater.util;

import java.util.Collection;
import java.util.HashSet;
import java.util.Iterator;
import java.util.Set;

import org.apache.wicket.extensions.markup.html.repeater.tree.ITreeProvider;
import org.apache.wicket.model.AbstractReadOnlyModel;
import org.apache.wicket.model.IDetachable;
import org.apache.wicket.model.IModel;

/**
 * A {@link Set} implementation utilizing a {@link ITreeProvider}'s models to keep containing
 * elements.
 * 
 * @author svenmeier
 * @param <T>
 *            type of data
 * 
 * @see ITreeProvider#model(Object)
 */
public class ProviderSubset<T> implements Set<T>, IDetachable
{

	private static final long serialVersionUID = 1L;

	private ITreeProvider<T> provider;

	private Set<IModel<T>> models = new HashSet<IModel<T>>();

	/**
	 * Create an empty subset.
	 * 
	 * @param provider
	 *            the provider of the complete set
	 */
	public ProviderSubset(ITreeProvider<T> provider)
	{
		this(provider, false);
	}

	/**
	 * Create a subset optionally containing all roots of the provider.
	 * 
	 * @param provider
	 *            the provider of the complete set
	 * @param addRoots
	 *            should all roots be added to this subset
	 */
	public ProviderSubset(ITreeProvider<T> provider, boolean addRoots)
	{
		this.provider = provider;

		if (addRoots)
		{
			Iterator<? extends T> roots = provider.getRoots();
			while (roots.hasNext())
			{
				add(roots.next());
			}
		}
	}

	@Override
	public void detach()
	{
		for (IModel<T> model : models)
		{
			model.detach();
		}
	}

	@Override
	public int size()
	{
		return models.size();
	}

	@Override
	public boolean isEmpty()
	{
		return models.size() == 0;
	}

	@Override
	public void clear()
	{
		detach();

		models.clear();
	}

	@Override
	public boolean contains(Object o)
	{
		IModel<T> model = model(o);

		boolean contains = models.contains(model);

		model.detach();

		return contains;
	}

	@Override
	public boolean add(T t)
	{
		return models.add(model(t));
	}

	@Override
	public boolean remove(Object o)
	{
		IModel<T> model = model(o);

		boolean removed = models.remove(model);

		model.detach();

		return removed;
	}

	@Override
	public Iterator<T> iterator()
	{
		return new Iterator<T>()
		{

			private Iterator<IModel<T>> iterator = models.iterator();

			private IModel<T> current;

			@Override
			public boolean hasNext()
			{
				return iterator.hasNext();
			}

			@Override
			public T next()
			{
				current = iterator.next();

				return current.getObject();
			}

			@Override
			public void remove()
			{
				iterator.remove();

				current.detach();
				current = null;
			}
		};
	}

	@Override
	public boolean addAll(Collection<? extends T> ts)
	{
		boolean changed = false;

		for (T t : ts)
		{
			changed |= add(t);
		}

		return changed;
	}

	@Override
	public boolean containsAll(Collection<?> cs)
	{
		for (Object c : cs)
		{
			if (!contains(c))
			{
				return false;
			}
		}
		return true;
	}

	@Override
	public boolean removeAll(Collection<?> cs)
	{
		boolean changed = false;

		for (Object c : cs)
		{
			changed |= remove(c);
		}

		return changed;
	}

	@Override
	public boolean retainAll(Collection<?> c)
	{
		throw new UnsupportedOperationException();
	}

	@Override
	public Object[] toArray()
	{
		throw new UnsupportedOperationException();
	}

	@Override
	public <S> S[] toArray(S[] a)
	{
		throw new UnsupportedOperationException();
	}

	@SuppressWarnings("unchecked")
	private IModel<T> model(Object o)
	{
		return provider.model((T)o);
	}

	/**
	 * Create a model holding this set.
	 * 
	 * @return model
	 */
	public IModel<Set<T>> createModel()
	{
		return new AbstractReadOnlyModel<Set<T>>()
		{
			/**
			 */
			private static final long serialVersionUID = 1L;

			@Override
			public Set<T> getObject()
			{
				return ProviderSubset.this;
			}

			@Override
			public void detach()
			{
				ProviderSubset.this.detach();
			}
		};
	}
}