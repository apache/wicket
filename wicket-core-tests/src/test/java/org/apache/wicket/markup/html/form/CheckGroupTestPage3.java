/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.wicket.markup.html.form;

import java.util.Arrays;

/**
 * tests rendering of second checkbox selected
 * 
 * @author igor
 * 
 */
public class CheckGroupTestPage3 extends AbstractCheckGroupTestPage
{
	/**
	 * Constructor
	 */
	public CheckGroupTestPage3()
	{
		super(Arrays.asList("check2"));

	}

	private static final long serialVersionUID = 1L;


}
