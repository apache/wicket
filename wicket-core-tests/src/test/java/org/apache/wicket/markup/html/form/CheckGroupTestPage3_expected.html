<html>
<head><title>CheckGroupTestPage3</title></head>
<body>
	<form wicket:id="form" id="form7" method="post" action="./org.apache.wicket.markup.html.form.CheckGroupTestPage3?2-1.-form"><div id="form7_hf_0" hidden="" class="hidden-fields"></div>
		
			<input type="checkbox" wicket:id="check1" id="check18" name="group" value="check0">check1</input>
			<span wicket:id="container">
				<input type="checkbox" wicket:id="check2" id="check29" name="group" value="check1" checked="checked">check2</input>
			</span>
		
	</form>
</body>
</html>
